UnitTests/UnitTests.HighPerformance.Shared/Enumerables/Test_ReadOnlyRefEnumerable{T}.cs

```diff
@@ -47,6 +47,53 @@ public void Test_ReadOnlyRefEnumerable_DangerousCreate_BelowZero(int length, int
         {
             _ = ReadOnlyRefEnumerable<int>.DangerousCreate(in Unsafe.NullRef<int>(), length, step);
         }
+
+        [TestCategory("ReadOnlyRefEnumerable")]
+        [TestMethod]
+        [DataRow(1, 1, new[] { 1 })]
+        [DataRow(4, 1, new[] { 1, 2, 3, 4 })]
+        [DataRow(4, 4, new[] { 1, 5, 9, 13 })]
+        [DataRow(4, 5, new[] { 1, 6, 11, 16 })]
+        public void Test_ReadOnlyRefEnumerable_Indexer(int length, int step, int[] values)
+        {
+            Span<int> data = new[]
+            {
+                1, 2, 3, 4,
+                5, 6, 7, 8,
+                9, 10, 11, 12,
+                13, 14, 15, 16
+            };
+
+            ReadOnlyRefEnumerable<int> enumerable = ReadOnlyRefEnumerable<int>.DangerousCreate(in data[0], length, step);
+
+            for (int i = 0; i < enumerable.Length; i++)
+            {
+                Assert.AreEqual(enumerable[i], values[i]);
+            }
+        }
+
+#if NETCOREAPP3_1_OR_GREATER
+        [TestCategory("ReadOnlyRefEnumerable")]
+        [TestMethod]
+        [DataRow(1, 1, new[] { 1 })]
+        [DataRow(4, 1, new[] { 1, 2, 3, 4 })]
+        [DataRow(4, 4, new[] { 1, 5, 9, 13 })]
+        [DataRow(4, 5, new[] { 1, 6, 11, 16 })]
+        public void Test_ReadOnlyRefEnumerable_Index_Indexer(int length, int step, int[] values)
+        {
+            Span<int> data = new[]
+            {
+                1, 2, 3, 4,
+                5, 6, 7, 8,
+                9, 10, 11, 12,
+                13, 14, 15, 16
+            };
+
+            ReadOnlyRefEnumerable<int> enumerable = ReadOnlyRefEnumerable<int>.DangerousCreate(in data[0], length, step);
+
+            Assert.AreEqual(values[^1], enumerable[^1]);
```

Comment:
I'm confused by this test (and same for the one in the other file), is this a copy-paste?
You seem to be reusing the same inputs as the other indexer test, but then you're not really using that data entirely, and the indexer is only ever called with `^1` once in the test method. Could you change this to do a loop to test the index from end in all positions, just for good measure? For instance, something like:

```csharp
for (int i = 1; i <= enumerable.Length; i++)
{
    Assert.AreEqual(values[^i], enumerable[^i]);
}
```

Reply:
> I'm confused by this test (and same for the one in the other file), is this a copy-paste?

It's shameful to admit, but my test-writing skills are pretty rusty. I followed the existing ones very closely, so they ended up basically being copy-pastes in the end :smile: